### ISSUE TITLE
Hover labels across shared axes

### ISSUE THREAD

**Issue by chriddyp** [plotly/plotly.js#2114]
As reported by a Dash community user: https://community.plot.ly/t/two-graphs-one-hover/6400/5

They would like hover labels to appear on all traces across all y-axes with shared x-axes. Right now, they only appear in the subplot that you are hovering in.

**Comment by etpinard** [comment]
Yeah - maybe under a new `hovermode`.

In the meantime: https://plot.ly/javascript/hover-events/#coupled-hover-events

**Comment by rreusser** [comment]
If so, I think it would hinge on the `id` attribute.

**Comment by deechiw** [comment]
After some searching came across this [Subplots Hover problem with closest data](https://community.plot.ly/t/subplots-hover-problem-with-closest-data/856)

**Comment by deechiw** [comment]
Would it make a difference if the graphs were in separate Divs, as in[Hide/show graph based on drop down selections](https://community.plot.ly/t/hide-show-graph-based-on-drop-down-selections/4725). Will the hover still be applicable, actually a vertical line would do it better

**Comment by lordgordon** [comment]
👍 for this feature.

I'm trying to implement a workaround using the "coupled hover events", yet so far I was unable to have it working.

Here my attempt so far: https://codepen.io/anon/pen/MXMqQa

**Comment by eddy-geek** [comment]
@lordgordon here is a [working pen](https://codepen.io/duyentnguyen/pen/LRVbyY) from the community forums

**Comment by eddy-geek** [comment]
So, after digging a bit, here is a solution that works for me, based on great work by others in the community.

The main tricks are
* dynamically get plot names
* use [visdcc.Runjs](https://github.com/jimmybow/visdcc#3-visdccrun_js-) to reload the javasript on graph change, to reattach the event handler to the re-created plot. *(other approaches like `setTimeout` and [dash_defer_js_import](https://github.com/zouhairm/dash_defer_js_import) work the first time only)*

```python
JS_STR = '''

var plotid = 'theplotname'
var plot = document.getElementById(plotid)

plot.on(
  'plotly_hover',
  function (eventdata) {
    console.log(eventdata.xvals);
    Plotly.Fx.hover(
      plotid, { 
        xval: eventdata.xvals[0] },
      Object.keys(plot._fullLayout._plots) // ["xy", "xy2", ...]
    );
  });
'''

layout = html.Div([
    dcc.Graph(id='theplotname'),
    visdcc.Run_js(id='hover-js')  # <-- add this
])

@app.callback(
    [Output('theplotname', 'figure'),
     Output('hover-js', 'run'),  # <-- add this
    ], [... inputs...])
def foo(inputs):
    figure = ...
    return figure, JS_STR  # <-- add this
```

**Comment by diggy2930** [comment]
**hello,
Can you  please copy a working example with datas, here my code:**

import dash
import dash_core_components as dcc
import dash_html_components as html
from dash.dependencies import Input, Output, State
import visdcc
import pandas as pd
import plotly.graph_objs as go
from _plotly_future_ import v4_subplots
from plotly.subplots import make_subplots
from dfply import *
from plotly import __version__
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot
import cufflinks as cf


from plotly import graph_objs as go
JS_STR = '''
var names = ['A', 'B', 'C']
var trace1 = {
  	x: [4, 5, 6],
 	y: [1, 2, 3],
  	text: names,
  	hoverinfo: 'x+y+text',
  	xaxis: 'x',
  	yaxis: 'y',
  	mode: 'markers',
  	type: 'scatter'
};

var trace2 = {
  	x: [50, 60, 70],
  	y: [1, 2, 3],
  	text: names,
  	hoverinfo: 'x+y+text',
  	xaxis: 'x2',
  	yaxis: 'y2',
  	mode: 'markers',
  	type: 'scatter'
};

var data = [trace1, trace2];

var layout = {
	hovermode:'compare',
	yaxis: {anchor: 'x'},
	xaxis: {domain: [0, 0.45]},
  	yaxis2: {anchor: 'x2'},
  	xaxis2: {domain: [0.55, 1]}

};

Plotly.newPlot('graph', data, layout);
var myPlot = document.getElementById('graph');
myPlot.on('plotly_hover', function (eventdata){
	console.log(eventdata.xvals);
    Plotly.Fx.hover('graph',
    				[
	    				{ curveNumber: 0, pointNumber:eventdata.points[0].pointNumber },
    					{ curveNumber: 1, pointNumber:eventdata.points[0].pointNumber },
    				],
        			['xy', 'x2y2']
    );
});
'''

app1 = dash.Dash(__name__)


        
app1.layout = html.Div([
    dcc.Graph(id='graph',figure),
    visdcc.Run_js(id='hover-js')
])



@app1.callback(
    [
     Output('graph','figure'),
     Output('hover-js', 'run'),  # <-- add this
    ])
def foo():
    figure=fig
    fig.update_layout(hovermode='x')   # "compare"
    return figure, JS_STR  # <-- add this
if __name__ == '__main__':
    app1.run_server()

**Comment by LukaPitamic** [comment]
Any way of achieving with python API?

**Comment by fmgithub2017** [comment]
@LukaPitamic , it would be great for R plotly too, it is a very nice feature.

**Comment by LukaPitamic** [comment]
@fmgithub2017 totally agree, I was trying to find a way around it but could not find any good solution

**Comment by jackparmer** [comment]
This issue has been tagged with `NEEDS SPON$OR`

A [community PR](https://github.com/plotly/plotly.js/pulls?q=is%3Apr+is%3Aopen+label%3A%22type%3A+community%22) for this feature would certainly be welcome, but our experience is deeper features like this are difficult to complete without the Plotly maintainers leading the effort.

**Sponsorship range: $15k-$20k**

What Sponsorship includes:
- Completion of this feature to the Sponsor's satisfaction, in a manner coherent with the rest of the Plotly.js library and API
- Tests for this feature
- Long-term support (continued support of this feature in the latest version of Plotly.js)
- Documentation at [plotly.com/javascript](https://plotly.com/javascript/)
- Possibility of integrating this feature with Plotly Graphing Libraries (Python, R, F#, Julia, MATLAB, etc)
- Possibility of integrating this feature with Dash
- Feature announcement on [community.plotly.com](https://community.plotly.com/tag/announcements) with shout out to Sponsor (or can remain anonymous)
- Gratification of advancing the world's most downloaded, interactive scientific graphing libraries (>50M downloads across supported languages)

Please include the link to this issue when [contacting us](https://plotly.com/get-pricing/) to discuss.

**Comment by immartian** [comment]
here's a good example, but wonder how it's possible to Python API so to speak

https://covid19-projections.com/path-to-herd-immunity/

**Comment by 1081** [comment]
This seems to work somehow. But is this possible with the Python API?
https://codepen.io/maio93/pen/OQxoZx

**Comment by LukaPitamic** [comment]
@Hans-Christian-1995 I looked into it, but couldn't find a way to do it with Python API method. Since we are doing everything in JLab JS method was a no-go. But this feature would make so much sense for us.

**Comment by Hans-Christian-1995** [comment]
@LukaPitamic , thank you for checking :). I agree, this feature would be very nice to have in Plotly. Is there any ways this can be developed?

**Comment by eddy-geek** [comment]
@Hans-Christian-1995 my [comment](https://github.com/plotly/plotly.js/issues/2114#issuecomment-535259328) above was for python (which this is probably not the right place to discuss). Unfortunately I think the solution stopped working with more recent versions of plotly, but it's certainly a good starting point.

**Comment by maguerreberry** [comment]
Hi, It'll be great to have this feature for python, any updates?

**Comment by AlanCPSC** [comment]
This addition would be fantastic! It would be much appreciated if the Plotly team could consider incorporating it into their roadmap for 2024. 👍

**Comment by LukaPitamic** [comment]
I totally agree, keep the good work!

**Comment by archmoj** [comment]
This would be resolved by #6947.

**Comment by LukaPitamic** [comment]
awesome!!!
can't wait
